src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+enum Shape {
+  'Triangle' = 'triangle',
+  'Circle' = 'circle',
+  'Rectangle' = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sorted: number[] = [a, b, c].sort((x: number, y: number) => x - y);
 
-export class Circle implements Figure {}
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw Error('Side length is 0 or less');
+    }
+
+    if (sorted[0] + sorted[1] <= sorted[2]) {
+      throw Error('Sides do not form a valid triangle');
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c)) * 100,
+      ) / 100
+    );
+  }
+}
+
+export class Circle implements Figure {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw Error('Radius must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius * this.radius * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (this.width <= 0 || this.height <= 0) {
+      throw Error('Width and height must be greater than 0');
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message. For example, 'Both width and height must be greater than 0 to form a valid rectangle'.